src/codewhisperer/service/inlineCompletion.ts

```diff
@@ -378,20 +379,36 @@ export class InlineCompletion {
             if (delay < showSuggestionDelay) {
                 this._timer?.refresh()
             } else {
-                // do not show recommendation if cursor is before invocation position
-                // and cancel paginated request
-                if (editor.selection.active.isBefore(RecommendationHandler.instance.startPos)) {
+                // do not show recommendation if cursor is before invocation position or user opened another document
+                // mark suggestions as Discard and cancel paginated request
+                if (
+                    editor.selection.active.isBefore(RecommendationHandler.instance.startPos) ||
+                    editor.document.uri.fsPath !== this.documentUri?.fsPath
+                ) {
+                    RecommendationHandler.instance.cancelPaginatedRequest()
+                    RecommendationHandler.instance.recommendations.forEach((r, i) => {
+                        RecommendationHandler.instance.setSuggestionState(i, 'Discard')
+                    })
```

Comment:
When would you want to set all the recommendation states to discard but not report and clear them?

Reply:
Hi Jaden, the state is marked as discard in the timer. But the function `getPaginatedRecommendation` is still running and when it sees the request is cancelled it will report telemetry. Thanks!